src/main/scala/tilelink/Xbar.scala

```diff
@@ -186,56 +190,105 @@ object TLXbar
-      (edgesOut zip reachableO).map { case (mp, reachable) =>
-        reachable && cp.client.anySupportProbe && mp.manager.managers.exists(_.regionType >= RegionType.TRACKED)
-      }.toVector}.toVector
-    val releaseIO = (edgesIn zip reachableIO).map { case (cp, reachableO) =>
-      (edgesOut zip reachableO).map { case (mp, reachable) =>
-        reachable && cp.client.anySupportProbe && mp.manager.anySupportAcquireB
-      }.toVector}.toVector
-
+    // "client" is master, "manager" is slave.
+    // Not every master need connect to every slave on every channel; determine which connections are necessary. It will generate a 2 dimensional array interconnect matrix. which represents the connectability of a master port to a slave port.
+    val reachableIO = edgesIn.map {
+      // Each parameters from master.
+      cp => edgesOut.map { 
+        // Each parameters from managers (sinks).
+        mp => 
+          cp
+          // Master port parameters.
+          .client
+          // All master parameters on the port.
+          .clients
+          .exists { 
+            c => 
+              mp
+              // Slave port parameters.
+              .manager
+              // All slave parameters on the port.
+              .managers
+              .exists {
+                m =>
+                  // fill true in the matrix, if master address sets have overlaps to any address of a slave on a port.
+                  c.visibility.exists { ca => m.address.exists { ma => ca.overlaps(ma)}}
+              }
+          }
+      }.toVector
+    }.toVector
+    // Another interconnect matrix, support slave probe master.
+    val probeIO = 
+      // Using zip to match each [[edgesIn]], [[edgesOut]] and [[reachableIO]], [[cp]], [[mp]] and [[reachable]] represents if this pair is connectable.
+      (edgesIn zip reachableIO).map {
+        case (cp, reachableO) =>
+          (edgesOut zip reachableO).map {
+            case (mp, reachable) =>
+              // Fill true in the matrix, if exist any master support probe, and exist any slave can probe.
+              reachable && cp.client.anySupportProbe && mp.manager.managers.exists(_.regionType >= RegionType.TRACKED)
+        }.toVector
+      }.toVector
+    // Another interconnect matrix, support slave probe master. support master release data.
+    val releaseIO = 
+      // Using zip to match each [[edgesIn]], [[edgesOut]] and [[reachableIO]],[[cp]], [[mp]] and [[reachable]] represents if this pair is connectable.
+      (edgesIn zip reachableIO).map { 
+        case (cp, reachableO) =>
+          (edgesOut zip reachableO).map { 
+            case (mp, reachable) =>
+              // Fill true in the matrix, if exist any master support probe, and exist any slave can probe.            
+              reachable && cp.client.anySupportProbe && mp.manager.anySupportAcquireB
+        }.toVector
+      }.toVector
+
+    // Interconnect matrix of A channel from master to slave.
```

Comment:
I think this is self-explanatory and if a comment is necessary, I'd rather have one then one for each val.